### PR TITLE
Fixed HDFFV-11150

### PR DESCRIPTION
**Pull request by bmribler**
Description
    Replaced an HDassert with a check for null pointer in H5O_dec_rc() to
    catch null pointer in corrupted data situation.
Platforms tested:
    Linux/64 (jelly)

### CODE REVIEW

**Comment by qkoziol** [comment]
Seems sensible - did you have a test for it?

**Comment by bmribler** [comment]
> Seems sensible - did you have a test for it?

The user (Mathworks) is still working on the permission to provide the corrupted file.  I couldn't craft a file that would cause segfault on this.

File: tools/test/h5dump/testh5dump.sh.in [hunk 1]
```diff
@@ -1435,8 +1495,12 @@ TOOLTEST3 non_existing.ddl --enable-error-stack tgroup.h5 non_existing.h5
 # test to verify HDFFV-9407: long double full precision
 GREPTEST OUTTXT "1.123456789012345" t128bit_float.ddl -m %.35Lf t128bit_float.h5
 
+# test HDFFFV-10480, using corrupted file from a CVE group, before the fix,
+# h5dump would segfault.
+TOOLTEST_FAIL t10480H5O_fill.ddl t10480H5O_fill.h5
+
 # Clean up temporary files/directories
-CLEAN_TESTFILES_AND_TESTDIR
+#CLEAN_TESTFILES_AND_TESTDIR
```
Comment (byrnHDF) [inline tools/test/h5dump/testh5dump.sh.in:1503]: Did you mean to leave this commented?

File: tools/test/h5dump/CMakeTests.cmake [hunk 2]
```diff
@@ -1131,6 +1132,9 @@
   # test to verify HDFFV-9407: long double full precision
   ADD_H5_GREP_TEST (t128bit_float 1 "1.123456789012345" -m %.35Lf t128bit_float.h5)
 
+  # test to verify HDFFV-10480 CVE-2018-11206
+  ADD_H5_TEST(t10480H5O_fill.ddl 0 t10480H5O_fill.h5)
```
Comment (byrnHDF) [inline tools/test/h5dump/CMakeTests.cmake:1136]: t10480H5O_fill.h5 needs to be added to HDF5_REFERENCE_TEST_FILES list

File: tools/test/h5dump/testh5dump.sh.in (hunk 1 above)
Comment (bmribler) [reply tools/test/h5dump/testh5dump.sh.in:1503]: > Did you mean to leave this commented?

Of course not! :-)  However, I realized I accidentally combined the new commit with an existing PR again!  Is there anyway for me to retract so they can be in two separate PRs?

Comment (lrknox) [reply tools/test/h5dump/testh5dump.sh.in:1503]: Of course there is.  One possibility (of many): 
- close this PR
- create a new fork 
- create 2 new branches from hdf5_1_10
- copy changes for each PR to one of the branches
- create the PRs
- sooner or later delete one of the forks

Comment (bmribler) [reply tools/test/h5dump/testh5dump.sh.in:1503]: Haha... I was hoping for not that!! ;-)  I guess that's the most straightforward for me.  Thank you, Larry!
Oh, wait, that means I need to delete bmribler/hdf5?  I will lose all the related stuff, right?

**Comment by bmribler** [comment]
I'm closing this PR and will create two separate PRs.

**Pull request closed without merging**